cmd/migrate_cmd.go

```diff
@@ -31,76 +41,127 @@ func migrate(cmd *cobra.Command, args []string) {
 		globalConfig.DB.Driver = u.Scheme
 	}
 
-	log := logrus.New()
-
-	pop.Debug = false
-	if globalConfig.Logging.Level != "" {
-		level, err := logrus.ParseLevel(globalConfig.Logging.Level)
-		if err != nil {
-			log.Fatalf("Failed to parse log level: %+v", err)
-		}
-		log.SetLevel(level)
-		if level == logrus.DebugLevel {
-			// Set to true to display query info
-			pop.Debug = true
-		}
-		if level != logrus.DebugLevel {
-			var noopLogger = func(lvl logging.Level, s string, args ...interface{}) {
-				return
-			}
-			// Hide pop migration logging
-			pop.SetLogger(noopLogger)
-		}
+	source := &sqlmigrate.FileMigrationSource{
+		Dir: "migrations/",
 	}
 
-	deets := &pop.ConnectionDetails{
-		Dialect: globalConfig.DB.Driver,
-		URL:     globalConfig.DB.URL,
-	}
-	deets.Options = map[string]string{
-		"migration_table_name": "schema_migrations",
+	db, err := sql.Open("pgx", globalConfig.DB.URL)
+	if err != nil {
+		logrus.Fatalf("Failed to connect to the database: %v", err.Error())
 	}
+	defer db.Close()
 
-	db, err := pop.NewConnection(deets)
+	// create migrations table for sql-migrate
+	_, err = db.Exec(`create table if not exists migrations (
+		id text, 
+		applied_at timestamptz, 
+		primary key (id)
+	)`)
 	if err != nil {
-		log.Fatalf("%+v", errors.Wrap(err, "opening db connection"))
+		logrus.Fatalln(err)
 	}
-	defer db.Close()
 
-	if err := db.Open(); err != nil {
-		log.Fatalf("%+v", errors.Wrap(err, "checking database connection"))
+	// check if schema_migrations table exists
+	res, err := db.Query("select exists(select from pg_tables where schemaname = 'auth' and tablename = 'schema_migrations')")
```

Comment:
Can consider using http://go-database-sql.org/retrieving.html#single-row-queries since it seems like a single row is returned

Reply:
oo i wasn't aware of this, cool!